builtin/logical/database/path_rotate_credentials.go

```diff
@@ -88,10 +92,6 @@ func (b *databaseBackend) pathRotateRootCredentialsUpdate() framework.OperationF
 			delete(b.connections, name)
 		}()
 
-		// Take out the backend lock since we are swapping out the connection
-		b.Lock()
-		defer b.Unlock()
-
 		// Take the write lock on the instance
 		dbi.Lock()
 		defer dbi.Unlock()
```

Comment:
Would it make sense to also move the `dbi` lock/unlock up so that it's before the lambda defer func above? We're modifying `dbi` there so we should have that lock held before releasing it right?

Reply:
I don't think so because the race that was detected was on the `b.connections` map. I'll investigate further to be sure.

Reply:
Good catch! I don't think we strictly *need* to since the DBI instance is being removed and shouldn't be referenced by anything else at this point, but I'll move the dbi lock up just to be safe.